modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java

```diff
@@ -638,9 +639,29 @@ private void _validate(
 				"Invalid output type " + outputType);
 		}
 
-		ObjectValidationRuleEngine objectValidationRuleEngine =
-			_objectValidationRuleEngineRegistry.getObjectValidationRuleEngine(
-				companyId, engine);
+		ObjectValidationRuleEngine objectValidationRuleEngine = null;
+
+		try {
+			objectValidationRuleEngine =
```

Comment:
@pedro-oliveira446 I notice this method is used also in [here](https://github.com/liferay-objects/liferay-portal/blob/4ae37f8914b774c30b8cde156276ea27d1f12a5f/modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java#L397-L401).

Wouldn't this be a problem when commerce team uses it?

Reply:
The idea is not to validate at the time of creation, only at the time of validation execution. This is where the validation runs, so we need to validate here.

Reply:
The bug is occurring due to an issue with the service registration in serviceTracker, not because the engine doesn't exist.